backend/controllers/prisonerProfile/prisonerLocationHistory.js

```diff
@@ -12,7 +12,37 @@ const {
   app: { notmEndpointUrl: dpsUrl },
 } = require('../../config')
 
-module.exports = ({ prisonApi, logError }) => async (req, res) => {
+const fetchStaffName = (context, staffId, prisonApi) => {
+  return prisonApi
+    .getStaffDetails(context, staffId)
+    .then(staff => formatName(staff.firstName, staff.lastName))
+    .catch('No staff name found')
+}
+
+const fetchReasonDescription = (context, assignmentReasonCode, caseNotesApi) => {
+  return caseNotesApi
+    .getCaseNoteTypes(context)
+    .then(caseNoteTypes => caseNoteTypes.find(type => type.code === 'MOVED_CELL'))
+    .then(cellMoveTypes => {
+      return cellMoveTypes?.subCodes.map(subType => ({
+        value: subType.code,
+        text: subType.description,
+      }))
+    })
+    .then(cellMoveReasonRadioValues => cellMoveReasonRadioValues.find(record => record.value === assignmentReasonCode))
+    .then(assignmentReason => assignmentReason.text)
+    .catch('No reason description found')
```

Comment:
Oh I see. If `cellMoveReasonRadioValues` or `assignmentReason` isn't defined then it will throw an error but your catching it and translating it here. But losing the rest of the Error info.

Reply:
yes changed it to  `.catch(err => 'No reason description found')` to return a user-friendly message, is that okay or should log more info about the error?

Reply:
I understand that these errors only would occur if there was a programming error in the code, is that right?

```
const fetchStaffName = (context, staffId, prisonApi) =>
  prisonApi
    .getStaffDetails(context, staffId)
    .then(staff => formatName(staff.firstName, staff.lastName))
    .catch(err => 'No staff name found')

const fetchReasonDescription = (context, assignmentReasonCode, caseNotesApi) =>
  caseNotesApi
    .getCaseNoteTypes(context)
    .then(caseNoteTypes => caseNoteTypes.find(type => type.code === 'MOVED_CELL'))
    .then(cellMoveTypes => {
      return cellMoveTypes?.subCodes.map(subType => ({
        value: subType.code,
        text: subType.description,
      }))
    })
    .then(cellMoveReasonRadioValues => cellMoveReasonRadioValues.find(record => record.value === assignmentReasonCode))
    .then(assignmentReason => assignmentReason.text)
    .catch(err => 'No reason description found')

const fetchWhatHappened = (context, offenderNo, bookingId, bedAssignmentHistorySequence, prisonApi, whereaboutsApi) =>
  whereaboutsApi
    .getCellMoveReason(context, bookingId, bedAssignmentHistorySequence)
    .then(cellMoveReason => prisonApi.getCaseNote(context, offenderNo, cellMoveReason.cellMoveReason.caseNoteId))
    .then(caseNote => caseNote.text)
    .catch(err => 'Not entered')
```

Reply:
or networking issues of course